src/linagora.esn.calendar/app/calendar-configuration/calendar-configuration-tab-main/calendar-configuration-tab-main.controller.js

```diff
@@ -18,15 +18,18 @@ require('../../components/modals/calendar-delete-confirmation/calendar-delete-co
     CAL_DAV_PATH,
     calPathBuilder,
     calUIAuthorizationService,
-    calCalendarDeleteConfirmationModalService
+    calCalendarDeleteConfirmationModalService,
+    calCalendarSecretAddressConfirmationModalService
   ) {
     var self = this;
 
     self.$onInit = $onInit;
     self.openDeleteConfirmationDialog = openDeleteConfirmationDialog;
+    self.openGetSecretLinkFConfirmationDialog = openGetSecretLinkFConfirmationDialog;
     self.removeCalendar = removeCalendar;
     self.unsubscribe = unsubscribe;
     self.exportCalendar = exportCalendar;
+    self.createSecretLinkwithToken = createSecretLinkwithToken;
```

Comment:
There is no need for this.

Reply:
must be defined for test

Reply:
You don't need it defined for the tests. And you shouldn't do that most of the time. The code should not be changed only for the purpose of testing because it will cause confusion later on. For example, when you do this: `self.createSecretLinkwithToken = createSecretLinkwithToken;`, people expect that it will be used in the templates.

By the way, this is not the function you need to test. The function you need to test here is `openGetSecretLinkConfirmationDialog`. Right now with your tests someone can remove the `openGetSecretLinkConfirmationDialog` function and break the feature, but the tests will still pass.

So here's how it should be done:

With this part of code:

```js
function openGetSecretLinkFConfirmationDialog() {
  calCalendarSecretAddressConfirmationModalService(self.calendar, createSecretLinkwithToken);
}
```

You can do something like: 

```
calCalendarDeleteConfirmationModalServiceMock = sinon.stub();

// ...

$provide.value('calCalendarDeleteConfirmationModalServiceMock', calCalendarDeleteConfirmationModalServiceMock)

// And in the tests:

expect(calCalendarDeleteConfirmationModalServiceMock).to.have.been.calledOnce;
expect(calCalendarDeleteConfirmationModalServiceMock.getCall(0).args[0]).to.equal(CalendarConfigurationTabMainController.calendar);

const createSecretLinkWithTokenFn = calCalendarDeleteConfirmationModalServiceMock.getCall(0).args[1];

createSecretLinkWithTokenFn();

expect(calendarServiceMock.generateTokenForSecretLink).to.have.been.calledWith(...);

//
```

Perhaps we can do a pair programming session after the daily today.